cost/forecasting/straight_line_forecast/linear_regression/straight_line_forecast_linear_regression.pt

```diff
@@ -249,15 +265,28 @@ script "js_make_data_package", type: "javascript" do
     "Category": "category",
     "Region": "region",
     "Service": "service",
-    "Vendor Account": "vendor_account"
+    "Vendor Account Name": "vendor_account_name"
+    "Billing Center": "billing_center_name"
+  }
+
+  //If param_dimension == "Billing Center", add Billing Center Name field to data
+  if (param_dimension == "Billing Center"){
+    _.each(ds_costs, function(cost) {
+      var matched_billing_center = _.find(ds_billing_centers, function(bc) { return cost.billing_center_id == bc.id })
+      if (matched_billing_center != undefined) {
+        cost["billing_center_name"] = matched_billing_center.name
+      } else {
+        cost["billing_center_name"] = cost.billing_center_id
+      }
+    })
   }
 
   //Ignore costs where Category is "Commitments"
   ds_costs = _.reject(ds_costs, function(cost){ return cost.category == "Commitments" })
 
   //For each month, for each type in dimension, sum costs
-  month = _.pluck(_.uniq(ds_costs, function(cost){ return cost.start_date }), "start_date" )
-  dimension = _.pluck(_.uniq(ds_costs, function(cost){ return cost[dimension_type[param_dimension]] }), dimension_type[param_dimension] )
+  var month = _.pluck(_.uniq(ds_costs, function(cost){ return cost.start_date }), "start_date" )
```

Comment:
Might be better to do pluck first, then uniq, just to make the code more clear and reduce its size. Example:

month = _.uniq(_.pluck(ds_costs, "start_date"))
dimension = _.uniq(_.pluck(ds_costs, dimension_type[param_dimension]))

Reply:
Done and retested.